### ISSUE TITLE
`axis` (and related) parameters for libraries without support for integer axis indices?

### ISSUE THREAD

**Issue by SimonHeybrock** [data-apis/array-api#252]
[scipp](https://scipp.github.io/) made the choice to not support integers for axis selection, but instead uses dimension labels (similar to xarray), since we considered integers too error prone.

While this is just one of the incompatibilities, let us for simplicity consider [Python array API: max](https://data-apis.org/array-api/latest/API_specification/statistical_functions.html#max-x-axis-none-keepdims-false). It defines:

```python
axis : Optional[ Union[ int, Tuple[ int, … ] ] ]
```

I failed to find any discussion on this, but has it been considered whether this could be made more flexible, e.g., to support

```python
axis : Optional[ Union[ int, Tuple[ int, … ], str, Tuple[ str, … ] ] ]
```
which would allow for selecting axes via dimension labels? I suppose one would require to support only *either* `int` *or* `str`, but not both.

**Comment by kgryte** [comment]
Selecting axes via dimension labels is probably a non-starter for the array API specification.

One of the goals of the specification is to promote compatibility across array libraries. If the specification were to add support for dimension labels, all aspiring specification-compliant array libraries (e.g., NumPy, Torch, TF, CuPy, etc) would need to support labels. I don't think this is likely.

Array libraries are certainly free to add support for labels in augmentation of the array API specification, but such feature support is not the lowest common denominator in this case.

You may want to follow the dataframe API standardization process, as similar discussions have been happening there wrt columns backed by multidimensional arrays.

**Comment by rgommers** [comment]
I agree with @kgryte. There's Xarray and scipp, and there's some experimental support for named axes in PyTorch, but that's about it - so it's not at a point where we can standardize anything. Also xref the TensorFlow feature request: https://github.com/tensorflow/tensorflow/issues/38228.

Do you mind if we close this @SimonHeybrock?

**Comment by SimonHeybrock** [comment]
Thanks for the references!
